src/EventTickets/Endpoints/ListEvents.php

```diff
@@ -0,0 +1,177 @@
+use WP_REST_Request;
+use WP_REST_Response;
+
+class ListEvents extends Endpoint
+{
+    /**
+     * @var string
+     */
+    protected $endpoint = 'admin/event-tickets';
+
+    /**
+     * @var WP_REST_Request
+     */
+    protected $request;
+
+    /**
+     * @var EventTicketsListTable
+     */
+    protected $listTable;
+
+    /**
+     * @inheritDoc
+     */
+    public function registerRoute()
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => 'GET',
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => [$this, 'permissionsCheck'],
+                ],
+                'args' => [
+                    'page' => [
+                        'type' => 'integer',
+                        'required' => false,
+                        'default' => 1,
+                        'minimum' => 1
+                    ],
+                    'perPage' => [
+                        'type' => 'integer',
+                        'required' => false,
+                        'default' => 30,
+                        'minimum' => 1
+                    ],
+                    'search' => [
+                        'type' => 'string',
+                        'required' => false
+                    ],
+                    'sortColumn' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortDirection' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'enum' => [
+                            'asc',
+                            'desc'
+                        ],
+                    ],
+                    'locale' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'default' => get_locale(),
+                    ],
+                    'return' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'default' => 'columns',
+                        'enum' => [
+                            'model',
+                            'columns'
+                        ],
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function handleRequest(WP_REST_Request $request): WP_REST_Response
+    {
+        $this->request = $request;
+        $this->listTable = give(EventTicketsListTable::class);
+
+        $events = $this->getEvents();
+        $eventsCount = $this->getTotalEventsCount();
+        $pageCount = (int)ceil($eventsCount / $request->get_param('perPage'));
+
+        if ('model' === $this->request->get_param('return')) {
```

Comment:
Instead of supporting a return type, I would suggest calling a dedicated route with the different return value/format.

Reply:
I'll refactor the Routes thing, but then take a Friday to refactor on the other domains.

Reply:
Resolved: 6374fcd8c3f55cc057803ae11fb05336ec24e4c2